### PR TITLE
Redo README.md

### PR DESCRIPTION
**Pull request by gilligan**
This is an attempt at providing a more helpful `README.md` which provides:

- CI status info
- key features
- where to find more info
- how to install
- how to develop
- how to contribute
- LICENSE

This should address most of #1304

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,38 +1,67 @@
...
+- **Extensible**: _NixOps_ is extensible through a plugin infrastructure which can be used to provide additional backends.
 
-To start developing on nixops, you can run:
+For more information please refer to the [NixOps manual](https://nixos.org/nixos/manual/).
 
-```bash
-  $ ./dev-shell
+### Installing
+
+_NixOps_ is included in nixpkgs and can be installed from your respective channel:
```
Comment (grahamc) [inline README.md:17]: We might should warn that this repo contains NixOps 2.0 and the channels don't package it yet?

File: README.md [hunk 2]
```diff
@@ -1,38 +1,67 @@
...
+
+```
+$ nix-env -i nixops
```
Comment (grahamc) [inline README.md:20]:
```suggestion
$ nix-shell -p nixops
```

File: README.md [hunk 3]
```diff
@@ -1,38 +1,67 @@
...
+
+```
+$ nix-env -if https://github.com/NixOs/nixops/tarball/master
+```
+
+### Building And Developing
+
+#### Building The Nix Package
+
+You can build the nix package by simply invoking `nix-build` on the project root:
```
Comment (grahamc) [inline README.md:33]:
```suggestion
You can build the Nix package by simply invoking `nix-build` on the project root:
```

File: README.md (hunk 1 above)
Comment (gilligan) [reply README.md:17]: Oh, yes indeed I should add something to that effect indeed.

File: README.md (hunk 2 above)
Comment (gilligan) [reply README.md:20]: I'm totally fine with changing this to `nix-shell` but then the description shouldn't be saying "installing" right? Then it should say something about "trying it in a shell" right?

File: README.md (hunk 1 above)
Comment (gilligan) [reply README.md:17]: So I added a Note below, let me know if that's OK for you.

File: README.md (hunk 2 above)
Comment (grahamc) [reply README.md:20]: "Use it in a shell" maybe? I've never installed nixops :)

Comment (grahamc) [reply README.md:20]: Yeah, let's stick to `nix-shell` for here.

Comment (gilligan) [reply README.md]: Alrighty, i changed the text above..

**Comment by grahamc** [comment]
Thanks!

**Pull request merged**